src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1460,13 +1460,27 @@ public static BigInteger Pow(BigInteger value, int exponent)
             if (bits == null)
                 return sign;
 
-            ulong h = bits[bits.Length - 1];
-            ulong m = bits.Length > 1 ? bits[bits.Length - 2] : 0;
-            ulong l = bits.Length > 2 ? bits[bits.Length - 3] : 0;
+            int length = bits.Length;
+
+            // Maximum exponent for double is 1023, uint bit length 32 => all bigint with bits[] longer 32 is infinity
```

Comment:
Could you reword this comment slightly, it's a bit confusing as written. Making the second line a separate sentence would help:

```
// The maximum exponent for doubles is 1023, which corresponds to a uint bit length of 32.
// All BigIntegers with bits[] longer than 32 evaluate to Double.Infinity (or NegativeInfinity).
// Cases where the exponent is between 1024 and 1035 are handled in NumericsHelpers.GetDoubleFromParts.
```